### ISSUE TITLE
Nodepool rolling upgrade management

### ISSUE THREAD

**Issue by LisaPerrier** [ovh/public-cloud-roadmap#462]
As a Managed Kubernetes customer,
I want to be able to define the number of maximum parallel installation that can be done in a nodepool.
So that each time a working node is updated, another update is started to optimize resource and time

**Comment by antonin-a** [comment]
Initial description edited in order to add node max surge and maintenance windows

**Comment by pyverdon** [comment]
Hello,
please consider adding those two nodepool upgrade enhancements:
- ability to run the upgrade of each nodepool in parallel ( this will speed up the upgrade for customer )
- keep the nodepool upgrade on hold if a Pod Disruption Budget constrain is not reached or have a configurable timeout.
-- current default respect PDBs for 10min only before moving on: customer should be able to configure this value and / or able to choose to have the upgrade on hold in such case

Thanks
